client/app/bundles/course/assessment/pages/AssessmentGenerate/utils.ts

```diff
@@ -37,111 +54,90 @@ export function extractQuestionPrototypeData(
           response.resources[0]?.solutions[0]?.files[0]?.content,
         ),
         testCases: {
-          public:
-            response?.resources?.[0]?.exprTestcases
-              ?.filter((testCase) => testCase?.visibility === 'public')
-              ?.map((testCase) => ({
-                expression: testCase.expression,
-                expected: 'True',
-                hint: testCase.hint,
-              })) ?? [],
-          private:
-            response?.resources?.[0]?.exprTestcases
-              ?.filter((testCase) => testCase?.visibility === 'private')
-              ?.map((testCase) => ({
-                expression: testCase.expression,
-                expected: 'True',
-                hint: testCase.hint,
-              })) ?? [],
-          evaluation:
-            response?.resources?.[0]?.exprTestcases
-              ?.filter((testCase) => testCase?.visibility === 'hidden')
-              ?.map((testCase) => ({
-                expression: testCase.expression,
-                expected: 'True',
-                hint: testCase.hint,
-              })) ?? [],
+          public: buildTestCases('public', response),
+          private: buildTestCases('private', response),
+          evaluation: buildTestCases('hidden', response),
         },
       },
     },
   };
 }
 
+type IndexedTestCase = Record<
+  number,
+  { expression: string; expected: string; hint: string }
+>;
+
 export const buildGenerateRequestPayload = (
   codaveriData: CodaveriGenerateFormData,
   questionData: QuestionPrototypeFormData,
 ): FormData => {
   const data = new FormData();
-  // TODO: Currently we are injecting the existing question data into the custom prompt directly.
-  // When Codaveri implements this as a feature, make sure to use the updated request model.
-  const fragments = [codaveriData.customPrompt];
+  const isFirstGeneratedQuestion =
+    JSON.stringify(questionData) === JSON.stringify(defaultQuestionFormData);
```

Comment:
Minor nit: this isn't so much the `firstGeneratedQuestion` as much as `isDefaultQuestionFormData`.

Reply:
done